### DISCUSSION TITLE
A Question About Intersect

### DISCUSSION THREAD

**Discussion by dxinliyong** [locationtech/jts#1097] [Q&A] [answered]
Hello Everyone,I have a question I would like to discuss with you.If you know, please don't hesitate to answer
The question as follows:
I have two layers, you can see the first layer(one.shp) is light green and the second layer(two.shp) is a transparent black frame and the four faces of the second layer overlap each other. Using ESRI OVERLAYER tools to find the intersection of two layers, I got 27 results(result.shp). The test data is included in the data.zip attachment.
![image](https://github.com/user-attachments/assets/3402e7da-5aa5-4498-babd-671571aeba41)
For example, the intersection of the four faces in the middle retains four copies. I know that this preserves the attributes of each face, but I don't know how its graphics are calculated.How to split the intersection into individual shapes and retain the properties of each shape.
![image](https://github.com/user-attachments/assets/1523b38b-ab38-4cfe-9f08-51168a169ef8)

[data.zip](https://github.com/user-attachments/files/17843610/data.zip)

**Comment by dr-jts** [comment, marked as answer]
See [`PolygonOverlayFunctions`](https://github.com/locationtech/jts/blob/master/modules/app/src/main/java/org/locationtech/jtstest/function/PolygonOverlayFunctions.java) for one way to do this.
